geomstats/visualization.py

```diff
@@ -342,28 +343,41 @@ def draw(self, ax, **kwargs):
 class PoincareHalfPlane:
     """Class used to plot points in the Poincare Half Plane."""
 
-    def __init__(self, points=None):
+    def __init__(self, points=None, point_type='half-plane'):
         self.points = []
+        self.point_type = point_type
         if points is not None:
             self.add_points(points)
 
     def add_points(self, points):
-        if not gs.all(H2.belongs(points)):
-            raise ValueError(
-                'Points do not belong to the hyperbolic space.')
-        points = self.convert_to_half_plane_coordinates(points)
+        if self.point_type == 'extrinsic':
+            if not gs.all(H2.belongs(points)):
+                raise ValueError(
+                    'Points do not belong to the hyperbolic space '
+                    '(extrinsic coordinates)')
+            points = self.convert_to_half_plane_coordinates(points)
+        elif self.point_type == 'half-plane':
+            if not gs.all(POINCARE_HALF_PLANE.belongs(points)):
+                raise ValueError(
```

Comment:
These lines are not covered by the unit-tests, maybe add a test with point_type=half-plane?

Reply:
(Have a look at https://sourcegraph.com/github.com/geomstats/geomstats to find out which lines are not being tested)